libs/cli/src/main/java/org/elasticsearch/cli/Terminal.java

```diff
@@ -285,6 +287,23 @@ static boolean isSupported() {
             return CONSOLE != null;
         }
 
+        static Console detectTerminal() {
+            // JDK >= 22 returns a console even if the terminal is redirected unless using -Djdk.console=java.base
+            // https://bugs.openjdk.org/browse/JDK-8308591
+            Console console = System.console();
+            if (console != null && Runtime.version().feature() >= JDK_VERSION_WITH_IS_TERMINAL) {
+                try {
+                    // verify the console is a terminal using isTerminal() on JDK >= 22
+                    // TODO: Remove reflection once Java 22 sources are supported, e.g. using a MRJAR
+                    Method isTerminal = Console.class.getMethod("isTerminal");
+                    return Boolean.TRUE.equals(isTerminal.invoke(console)) ? console : null;
+                } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+                    throw new RuntimeException(e);
```

Comment:
nit: I would make this AssertionError, if we hit this, something is very wrong (jdk22 is missing a method).